sync3/handler/connstate.go

```diff
@@ -590,10 +590,20 @@ func (s *ConnState) getInitialRoomData(ctx context.Context, roomSub sync3.RoomSu
 	}
 	roomToTimeline = s.userCache.AnnotateWithTransactionIDs(ctx, s.userID, s.deviceID, roomToTimeline)
 	rsm := roomSub.RequiredStateMap(s.userID)
+
+	// The room state is only needed for rooms we are actually joined to, so we don't need to fetch it
+	// for rooms we are invited to.
```

Comment:
Would probably be better to filter _out_ invites rather than filter _in_ joined rooms to keep banned/left rooms.